react/src/components/Table/DownloadModal.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from 'react';
+import { ColumnInstance, Row } from 'react-table';
+import { downloadCsv } from '../../utils';
+import { Button, Input, Modal } from '../index';
+import { InlineFlex } from '../Layout';
+
+interface DownloadModalProps<T extends object> {
+    rows: Row<T>[];
+    visibleColumns: ColumnInstance<T>[];
+}
+
+export default function DownloadModal<T extends {}>({
+    rows,
+    visibleColumns,
+}: DownloadModalProps<T>) {
+    const [filename, setFilename] = useState<string>('');
+    const [showDownload, setShowDownload] = useState<boolean>(false);
+
+    return (
+        <InlineFlex>
+            <Button variant="primary" onClick={() => setShowDownload(true)}>
+                Export Data
+            </Button>
+            <Modal
+                active={showDownload}
+                hideModal={() => {
+                    setShowDownload(false);
+                    setFilename('');
+                }}
+                title="Save Variants"
+                footer="Download CSV"
+                onClick={() =>
+                    downloadCsv(
+                        rows.map(r => r.values as T),
+                        visibleColumns
+                            .filter(c => c.id && !c.id.match(/^empty/i))
+                            .map(c => c.id as keyof T),
```

Comment:
Do you still need the typecasts when you've typed the `rows` argument as `Row<T>[]`?

Reply:
I changed the types to IdType (which is what react-table uses for its keyof T | string) type. We'd still need to cast it as keyof T in `downloadCsv.ts` but I think it's cleaner.